Bottom: f2598c6b666cf78709917cd58b6fa861d8435cdf
Top:    ec5db1ae1416c2def689d3a39e1cee5091831ef2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 22:55:32 +0000

Refresh of revert-stop-find-connected-fol

---

diff --git a/src/action.c b/src/action.c
index 7e5f6b1..0563c2b 100644
--- a/src/action.c
+++ b/src/action.c
@@ -874,7 +874,7 @@ NotifyLine (void)
 			       PIN_TYPE | PAD_TYPE | VIA_TYPE, &ptr1, &ptr2,
 			       &ptr3);
 	  LookupConnection (Crosshair.X, Crosshair.Y, true, 1,
-			    FOUNDFLAG, true);
+			    FOUNDFLAG);
 	}
       if (type == PIN_TYPE || type == VIA_TYPE)
 	{
@@ -2757,7 +2757,7 @@ ActionDisplay (int argc, char **argv, Coord childX, Coord childY)
 	      if (Crosshair.AttachedLine.State != STATE_FIRST)
 		LookupConnection (Crosshair.AttachedLine.Point1.X,
 				  Crosshair.AttachedLine.Point1.Y, true, 1,
-				  FOUNDFLAG, true);
+				  FOUNDFLAG);
 	    }
 	  notify_crosshair_change (true);
 	  break;
diff --git a/src/find.c b/src/find.c
index be9760e..e3d9316 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3103,8 +3103,7 @@ ListStart (int type, void *ptr1, void *ptr2, void *ptr3)
  * also the action is marked as undoable if AndDraw is true
  */
 void
-LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
-                  bool AndRats)
+LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag)
 {
   void *ptr1, *ptr2, *ptr3;
   char *name;
@@ -3118,9 +3117,10 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
     = SearchObjectByLocation (LOOKUP_FIRST, &ptr1, &ptr2, &ptr3, X, Y, Range);
   if (type == NO_TYPE)
     {
-      type = SearchObjectByLocation (
-        LOOKUP_MORE & ~(AndRats ? RATLINE_TYPE : 0),
-        &ptr1, &ptr2, &ptr3, X, Y, Range);
+      type
+        =
+        SearchObjectByLocation
+        (LOOKUP_MORE, &ptr1, &ptr2, &ptr3, X, Y, Range);
       if (type == NO_TYPE)
         return;
       if (type & SILK_TYPE)
@@ -3133,9 +3133,11 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
             return;
         }
     }
-
-  name = ConnectionName (type, ptr1, ptr2);
-  hid_actionl ("NetlistShow", name, NULL);
+  else
+    {
+      name = ConnectionName (type, ptr1, ptr2);
+      hid_actionl ("NetlistShow", name, NULL);
+    }
 
   TheFlag = which_flag;
   User = AndDraw;
@@ -3145,7 +3147,7 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
    * This is step (1) from the description
    */
   ListStart (type, ptr1, ptr2, ptr3);
-  DoIt (AndRats, AndDraw);
+  DoIt (true, AndDraw);
   if (User)
     IncrementUndoSerialNumber ();
   User = false;
diff --git a/src/find.h b/src/find.h
index 9644727..b54fd99 100644
--- a/src/find.h
+++ b/src/find.h
@@ -50,7 +50,7 @@ bool LinePadIntersect (LineType *, PadType *);
 bool ArcPadIntersect (ArcType *, PadType *);
 void LookupElementConnections (ElementType *, FILE *);
 void LookupConnectionsToAllElements (FILE *);
-void LookupConnection (Coord, Coord, bool, Coord, int, bool AndRats);
+void LookupConnection (Coord, Coord, bool, Coord, int);
 void LookupUnusedPins (FILE *);
 bool ResetFoundLinesAndPolygons (bool, int flag);
 bool ResetFoundPinsViasAndPads (bool, int flag);
diff --git a/src/netlist.c b/src/netlist.c
index 4a17eec..d47f7d2 100644
--- a/src/netlist.c
+++ b/src/netlist.c
@@ -164,7 +164,7 @@ netlist_find (LibraryMenuType * net, LibraryEntryType * pin)
   int x, y;
   if (pin_name_to_xy (net->Entry, &x, &y))
     return;
-  LookupConnection (x, y, 1, 1, FOUNDFLAG, true);
+  LookupConnection (x, y, 1, 1, FOUNDFLAG);
 }
 
 static void
@@ -173,7 +173,7 @@ netlist_select (LibraryMenuType * net, LibraryEntryType * pin)
   int x, y;
   if (pin_name_to_xy (net->Entry, &x, &y))
     return;
-  LookupConnection (x, y, 1, 1, SELECTEDFLAG, true);
+  LookupConnection (x, y, 1, 1, SELECTEDFLAG);
 }
 
 static void
diff --git a/src/report.c b/src/report.c
index a5e45de..6285758 100644
--- a/src/report.c
+++ b/src/report.c
@@ -560,7 +560,7 @@ XYtoNetLength (Coord x, Coord y, int *found)
   /* NB: The third argument here, 'false' ensures LookupConnection
    *     does not add its changes to the undo system.
    */
-  LookupConnection (x, y, false, PCB->Grid, FOUNDFLAG, true);
+  LookupConnection (x, y, false, PCB->Grid, FOUNDFLAG);
 
   ALLLINE_LOOP (PCB->Data);
   {
